homeassistant/components/fritz/common.py

```diff
@@ -257,6 +264,15 @@ def signal_device_update(self) -> str:
         """Event specific per FRITZ!Box entry to signal updates in devices."""
         return f"{DOMAIN}-device-update-{self._unique_id}"
 
+    def _set_guest_wifi_password(self, password: str, length: int) -> None:
+        """Set the guest networks wifi password."""
+        try:
+            self.fritz_guest_wifi.set_password(password, length)
+        except (FritzArgumentStringToShortError, FritzArgumentStringToLongError) as ex:
+            raise HomeAssistantError(
+                "Password must be between 8 and 63 characters length."
+            ) from ex
```

Comment:
you can add a validation schema while registering the service (see [`hass.services.async_register`](https://dev-docs.home-assistant.io/en/master/api/core.html?highlight=services.async_register#homeassistant.core.ServiceRegistry.async_register)), so invalid inputs are handled proper by HA itself

Reply:
I've tried this but can't get it to work. Maybe you could guide me.

I created this schema  in `conf.py`:
```
SERVICE_SCHEMA_GUEST_WIFI_PW_LENGTH = vol.Schema(
    {vol.Optional("length"): vol.Range(min=8, max=63)}
)
```

But I don't really understand how to use it and at which part. Must the service call be in a own function instead of `services_fritzbox`? And where to call `hass.services.async_register` than?

Reply:
the schema must contain all parameters and needs to be added during `hass.services.async_register()` - this is done here:
https://github.com/home-assistant/core/blob/09408234a6e07a0e6d6fb709f750dae9f9f69d65/homeassistant/components/fritz/services.py#L55

So the [`SERVICE_LIST`](https://github.com/home-assistant/core/blob/09408234a6e07a0e6d6fb709f750dae9f9f69d65/homeassistant/components/fritz/services.py#L21) needs to be extended, so that a schema can optionally be included.

Reply:
Thanks for the hint. Added a service schema now :)